widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt

```diff
@@ -409,7 +409,7 @@ internal class ChatController(
                 items.removeAll { (it as? VisitorAttachmentItem)?.messageId == message.id }
                 addVisitorAttachmentItemsToChatItems(items, message, currentMessageIndex + 1)
                 return@emitChatItems chatState.changeItems(items)
-            } else {
+            } else if (message.senderType == Chat.Participant.SYSTEM) {
```

Comment:
Two things seem unlogical here to me:
1. I would expect to see the `if (message.senderType == Chat.Participant.SYSTEM) ` check at the beginning of `onUnengagementMessage()` function. I expect this because the name of the function makes me think that everything that is inside the function should be related to UnengagementMessage. And `UnengagementMessage` is a message with `Chat.Participant.SYSTEM`, isn't it (maybe I'm mistaken in this statement)?
2. Do we still need an `else` case or no need to do anything if none of conditions are true?

Reply:
1. Unengagement Messages are all messages that come before an engagement started. We need it to display system messages and visitor attachments. But sometimes the engagement started event comes after the welcome message. It this case the operator message will come in the Unengagement Messages callback. To avoid adding this message in wrong order the condition was added.
2. No, we don't need to do anything else.

Reply:
Okay, got it.
Looks like I have an idea for a better name for `UnengagementMessage` - `PreengagementMessage`. 🙂 
However, it's too late, I know.

Reply:
@andrews-moc You're a genious!
I will rename yem!